docs: add a tip for disconnected event (#3353)
emqtt/emq-docs-cn: en_US/data-integration/rule-sql-events-and-fields.md

| `node`            | EMQX node where the event is triggered                       |
| `client_attrs`        | [Client attributes](../client-attributes/client-attributes.md) |

::: tip Building Client Presence State

> If you are building MQTT client presence state externally based on timestamps of disconnected and connected events, it's important to filter out disconnected events when the `reason` is either `takenover` or `discarded` (i.e., `WHERE reason != 'takenover' AND reason != 'discarded'`).

Reviewer: ```suggestion
If you maintain MQTT client presence state in an external system based on connection and disconnection event timestamps, filter out disconnection events where `reason` is `takenover` or `discarded` (i.e., `WHERE reason != 'takenover' AND reason != 'discarded'`)

```

> This is because the `disconnected_at` timestamp can be later than the `connected_at` timestamp in the event of a client reconnect. In EMQX versions before 5.9.0, `disconnected_at < connected_at` in most cases. Starting from EMQX 5.9.0, `disconnected_at > connected_at` in most cases. Filtering out these events ensures accurate presence state tracking.

Reviewer: ```suggestion
These events indicate that a new session with the same ClientID has replaced the previous one, rather than the client going offline.

Event timestamp behavior differs across EMQX versions:

- Before 5.9.0, `disconnected_at` is typically less than `connected_at`;
- Since 5.9.0, `disconnected_at` is typically greater than `connected_at`.

Relying solely on timestamp ordering may lead to incorrect presence calculations. Filtering these events ensures accurate presence tracking across versions.
```
